Mathlib/Algebra/Lie/Nilpotent.lean

```diff
@@ -716,6 +718,24 @@ theorem isNilpotent_of_le (M₁ M₂ : LieSubmodule R L M) (h₁ : M₁ ≤ M₂
   have hfg : ∀ x m, ⁅f x, g m⁆ = g ⁅x, m⁆ := by aesop
   exact (Submodule.inclusion_injective h₁).lieModuleIsNilpotent hfg
 
+/-- The max nilpotent submodule is the `sSup` of all nilpotent submodules. -/
+def maxNilpotentSubmodule :=
+  sSup { N : LieSubmodule R L M | IsNilpotent L N }
+
+instance maxNilpotentSubmoduleIsNilpotent [IsNoetherian R M] :
+    IsNilpotent L (maxNilpotentSubmodule R L M) := by
+  have hwf := LieSubmodule.wellFoundedGT_of_noetherian R L M
+  rw [← CompleteLattice.isSupClosedCompact_iff_wellFoundedGT] at hwf
+  refine hwf { N : LieSubmodule R L M | IsNilpotent L N } ⟨⊥, ?_⟩ fun N₁ h₁ N₂ h₂ => ?_
+  · simp [trivialIsNilpotent L]
+  · rw [Set.mem_setOf_eq] at h₁ h₂
+    apply instIsNilpotentAdd R L
```

Comment:
In this case the two goals are very close to being just instances, so we should try to avoid doing things manually. We can golf as follows:
```suggestion
  refine hwf { N : LieSubmodule R L M | IsNilpotent L N } ⟨⊥, ?_⟩ fun N₁ h₁ N₂ h₂ => ?_ <;>
  simp_all <;> infer_instance
```
provided we first make the following change higher up:
```diff
- instance instIsNilpotentAdd (M₁ M₂ : LieSubmodule R L M) [IsNilpotent L M₁] [IsNilpotent L M₂] :
-     IsNilpotent L (M₁ + M₂ : LieSubmodule R L M) := by
+ instance instIsNilpotentSup (M₁ M₂ : LieSubmodule R L M) [IsNilpotent L M₁] [IsNilpotent L M₂] :
+    IsNilpotent L (M₁ ⊔ M₂ : LieSubmodule R L M) := by
```